src/base_client.rs

```diff
@@ -328,6 +306,47 @@ impl Client {
         room.receive_state_event(event)
     }
 
+    /// Receive a presence event from an `IncomingResponse` and updates the client state.
+    ///
+    /// Returns true if the membership list of the room changed, false
+    /// otherwise.
+    ///
+    /// # Arguments
+    ///
+    /// * `room_id` - The unique id of the room the event belongs to.
+    ///
+    /// * `event` - The event that should be handled by the client.
+    pub fn receive_presence_event(&mut self, room_id: &str, event: &PresenceEvent) -> bool {
+        let user_id = &self
+            .session
+            .as_ref()
+            .expect("to receive events you must be logged in")
+            .user_id;
+        if self.current_room_id.comes_after(user_id, event) {
+            self.current_room_id.update(room_id, event);
+        }
+        // this should be guaranteed to find the room that was just created in the `Client::sync` loop.
```

Comment:
Guaranteed is a bit of strong word here. A server could send us presence info for bogus rooms, sure it could do so with other events but then we at least have something to show. So we might not want to create a room, just get and if it's None skip it.

Reply:
I was wondering about this in another branch I think I have a comment about making just a regular get method, does that sound good?

Reply:
Sounds good.